src/test/scala/mesosphere/marathon/tasks/TaskTrackerTest.scala

```diff
@@ -474,4 +477,50 @@ class TaskTrackerTest extends MarathonSpec with Matchers {
 
     verify(state, times(1)).update(any())
   }
+
+  // sounds strange, but this is how it currently works: checkStagedTasks will consider a missing startedAt to
+  // determine whether a task is in staging and might need to be killed if it exceeded the taskLaunchTimeout
+  test("ensure that determineOverdueTasks returns tasks disregarding the stagedAt property") {
+    import scala.language.implicitConversions
+    implicit def toMillis(timestamp: Timestamp): Long = timestamp.toDateTime.getMillis
+
+    val clock = ConstantClock(Timestamp.now())
+    val now = clock.now()
+
+    val unstagedTask = MarathonTask.newBuilder()
+      .setId("unstaged")
+      .build()
+
+    val stagedTask = MarathonTask.newBuilder()
+      .setId("staged")
+      .setStagedAt(now - 10.seconds)
+      .build()
+
+    val stagedLongTimeAgoTask = MarathonTask.newBuilder()
+      .setId("stagedLongTimeAgoTask")
+      // When using MarathonTasks.makeTask, this would be set to a made up value
+      // This test shall explicitly make sure that the task gets selected even if it is unlikely old
+      .setStagedAt(now - 10.days)
+      .build()
+
+    val runningTask = MarathonTask.newBuilder()
+      .setId("running")
+      .setStagedAt(now - 2.seconds)
+      .setStartedAt(now - 5.seconds)
+      .build()
+
+    Given("several tasks of which not all have been staged yet")
+    taskTracker.created(TEST_APP_NAME, unstagedTask)
+    taskTracker.created(TEST_APP_NAME, stagedLongTimeAgoTask)
+    taskTracker.created(TEST_APP_NAME, stagedTask)
+    taskTracker.created(TEST_APP_NAME, runningTask)
+
+    When("We check which tasks should be killed because they're not yet staged")
+    val stagedTasks = taskTracker.determineOverdueTasks(now)
+
+    Then("Both the unstaged and the stagedLongAgo tasks are returned")
+    assert(unstagedTask.getStagedAt == 0, "The stagedAt property of an unstaged task has a value of 0")
```

Comment:
startedAt instead of getStagedAt. I would use that assert after constructing the unstagedTask, because it is more a requirement of the test data than an expected behavior of your tested code.